lib/plugins/newrelicTransactionManagerPlugin.ts

```
@@ -30,29 +28,33 @@ export interface NewRelicTransactionManagerOptions {
 }
 
 export class NewRelicTransactionManager implements TransactionObservabilityManager {
+  public readonly newrelic?: Newrelic
+
   private readonly isEnabled: boolean
   private readonly transactionMap: FifoMap<TransactionHandle>
 
-  private constructor(isNewRelicEnabled: boolean) {
+  private constructor(isNewRelicEnabled: boolean, newrelic?: Newrelic) {
     this.isEnabled = isNewRelicEnabled
+    this.newrelic = newrelic
     this.transactionMap = new FifoMap(2000)
   }
 
   public static async create(isNewRelicEnabled: boolean): Promise<NewRelicTransactionManager> {
-    if (isNewRelicEnabled && !newrelic) {
+    let newrelic: Newrelic | undefined = undefined
+    if (isNewRelicEnabled) {
       // @ts-ignore
       newrelic = await import('newrelic')
     }
 
-    return new NewRelicTransactionManager(isNewRelicEnabled)
+    return new NewRelicTransactionManager(isNewRelicEnabled, newrelic)
   }
 
   public addCustomAttribute(attrName: string, attrValue: string | number | boolean) {
     if (!this.isEnabled) {
       return
     }
 
-    newrelic.addCustomAttribute(attrName, attrValue)
+    this.newrelic?.addCustomAttribute(attrName, attrValue)
```

shouldn't ever happen due to `isEnabled` flag, I'd say it's better to use `!` here and throw an error, otherwise we may silently lose observability, but we can do that later, and merge it as-is to unblock

---

Yeah, that is the idea, with this we at least unblock and avoid issues, and then we can move with `!` and check what happens